test/message/test_plan.py

```diff
@@ -431,10 +431,91 @@ def plan_mult_3(data):
     hook.local_worker.is_client_worker = True
 
 
+@pytest.mark.parametrize("is_func2plan", [True, False])
+def test_fetch_stateful_plan(hook, is_func2plan, workers):
+    hook.local_worker.is_client_worker = False
+
+    if is_func2plan:
+
+        @sy.func2plan(args_shape=[(1,)], state={"bias": th.tensor([3.0])})
+        def plan(data, state):
+            bias = state.read("bias")
+            return data * bias
+
+    else:
+
+        class Net(sy.Plan):
+            def __init__(self):
+                super(Net, self).__init__()
+                self.fc1 = nn.Linear(1, 1)
+                self.add_to_state(["fc1"])
+
+            def forward(self, x):
+                return self.fc1(x)
+
+        plan = Net()
+        plan.build(th.tensor([1.2]))
+
+    alice = workers["alice"]
+    sent_plan = plan.send(alice)
+
+    # Fetch plan
+    fetched_plan = alice.fetch_plan(sent_plan.id)
+
+    # Execute it locally
+    x = th.tensor([-1.26])
+    assert (fetched_plan(x) == sent_plan(x)).all()
```

Comment:
I wonder if plan.send() is inplace and if sent_plan == plan
I this is the case, then you can write `assert (fetched_plan(x) == plan(x)).all()`
If not, I would expect sent_plan(x) to return a ptr

Reply:
> I wonder if plan.send() is inplace and if sent_plan == plan

Yes they should be